cápsulas/cápsula_18/app.js

```diff
@@ -0,0 +1,81 @@
+const albumsContent = document.getElementById('albums-content')
+const templateAlbum = document.getElementById('template-album').content
+const wishlistContent = document.getElementById('wishlist-content')
+const templateWishlist = document.getElementById('template-wishlist').content
+
+const fragment = document.createDocumentFragment()
+
+let wishList = {}
+
+albumsContent.addEventListener('click', event => {
+  addToWishList(event)
+})

```

Comment:
Me parece que aquí estás agregándole el listener a todo el div `#albums-content`. Sería más preciso agregarle el listener al botón para agregar. Si bien estarías agregando un listener por cada album (que es perfectamente posible dentro del loop donde clonas el template por cada album), queda mucho más aislado que en el caso de que cualquier click dentro del div`#albums-content` capture el evento y ejecute `addToWishList`.

Reply:
He intentado arreglar esto, pero causa un conflicto con el evento DOMContentLoaded que impide que se carguen las tarjetas correspondientes a álbumes, lo que había tratado de hacer era iterar en un for y capturar todos los botones de este tipo de esta forma:
`const albumCardButtons = document.getElementsByClassName("btn-add")
for(var i=0; i< albumCardButtons.length; i++) {
  var currentButton = albumCardButtons[i]
  currentButton.addEventListener('click', event => {
    addToWishList(event);
  })
})`

Reply:
Claro, el tema con lo que intentaste es que los botones no existen inicialmente, sino que hasta que la función `getAlbums` termina de ejecutarse, y es como es asíncrona, siempre ocurrirá después de que se ejecutó todo el script. Lo que yo propongo es que el listener se agregue en el momento en que se construye la lista de álbumes dentro de la función `getAlbums`. Dejo a continuación el código asociado:

```javascript
// Esto se remueve completamente. Lo dejo como comentario para indicarlo, pero debería eliminarse
// albumsContent.addEventListener('click', event => {
//   addToWishList(event);
// })

const getAlbums = async () => {
  try {
    const res = await fetch('albums.json');
    const albums = await res.json();
    albums.forEach(album => {
      const clone = templateAlbum.cloneNode(true);
      const cloneBtnAdd = clone.querySelector('.btn-add');
      clone.querySelector('h3').textContent = album.name;
      clone.querySelector('p').textContent = album.artist;
      cloneBtnAdd.dataset.id = album.id;
      cloneBtnAdd.addEventListener('click', addToWishList); // Esta es la línea clave
      fragment.appendChild(clone);
    })
    albumsContent.appendChild(fragment);
  } catch (error) {
    console.log(error);
  }
}

const addToWishList = (event) => {
  setWishlist(event.target.parentElement);
}
```
Dejaré mi propuesta de código para el botón de eliminar de la wishlist en el comentario que dejé asociado a ese botón.

Reply:
Listo, agregué ambos cambios, muchas gracias!